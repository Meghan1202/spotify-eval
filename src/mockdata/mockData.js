const mockLike = { like: false, count: 1 };

const mockSongs = [{
  id: '1',
  name: 'song_name',
  genre: { id: '1', name: 'genre_name' },
  artist: { id: '1', name: 'abc' },
  albumArtUrl: 'abc.com',
  publishedAt: '',
  likes: 1,
  isLikedAlready: true,
}];

module.exports = { mockLike, mockSongs };
