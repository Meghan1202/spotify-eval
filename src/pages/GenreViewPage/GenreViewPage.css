.Genre-Page {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
}

.Genre-Container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}

.Genre-Page-Header {
    display: flex;
    justify-content: space-between;
    margin: 1rem 4rem
}