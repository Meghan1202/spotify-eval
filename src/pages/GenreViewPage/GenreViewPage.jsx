import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import GenreHeader from '../../components/GenreHeader/GenreHeader';
import SongCard from '../../components/SongCard/SongCard';
import './GenreViewPage.css';

const GenreViewPage = ({
  genreData, patchLike, updateSongs, toggleView,
}) => (
  <div className="Genre-Page">
    <div className="Genre-Page-Header">
      <h1>genres</h1>
      <Link
        to="/allSongs"
        onClick={() => {
          toggleView();
        }}
      >
        <img data-testid="Toggle-View" src="./assets/icon-grid.svg" alt="Toggle View" />
      </Link>
    </div>
    {
            Object.keys(genreData).map((genre) => (
              <React.Fragment key={genre}>
                <GenreHeader key={genre} Genre={genre} />
                <div className="Genre-Container">
                  {
                    genreData[genre].map((song, index) => (
                      <SongCard
                        key={song.id}
                        name={song.name}
                        albumArt={song.albumArtUrl}
                        artist={song.artist.name}
                        currentState={song.isLikedAlready}
                        id={song.id}
                        likes={song.likes}
                        patchLike={patchLike}
                        updateSongs={updateSongs}
                        theme={index % 2}
                      />
                    ))
                    }
                </div>
              </React.Fragment>
            ))
        }
  </div>
);

GenreViewPage.propTypes = {
  genreData: PropTypes.objectOf(Object).isRequired,
  patchLike: PropTypes.func.isRequired,
  updateSongs: PropTypes.func.isRequired,
  toggleView: PropTypes.func.isRequired,
};

export default GenreViewPage;
