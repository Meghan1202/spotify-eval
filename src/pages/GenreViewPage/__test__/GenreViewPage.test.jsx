import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import GenreViewPage from '../GenreViewPage';
import { mockGenre } from '../../../mockdata/mockData';

describe(GenreViewPage.name, () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <GenreViewPage
          genreData={mockGenre}
          patchLike={jest.fn()}
          updateSongs={jest.fn()}
          toggleView={jest.fn()}
        />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
