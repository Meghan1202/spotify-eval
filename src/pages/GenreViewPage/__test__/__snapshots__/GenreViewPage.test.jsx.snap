// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreViewPage should match the snapshot 1`] = `
<div>
  <div
    class="Genre-Page"
  >
    <div
      class="Genre-Page-Header"
    >
      <h1>
        genres
      </h1>
      <a
        href="/allSongs"
      >
        <img
          alt="Toggle View"
          src="./assets/icon-grid.svg"
        />
      </a>
    </div>
    <div
      class="Genre-Header-Container"
    >
      <img
        alt="genre"
        class="Genre-Image"
        src="./assets/genre-genre_name.png"
      />
      <div
        class="Genre-Indicator"
      >
        genre_name
      </div>
    </div>
    <div
      class="Genre-Container"
    >
      <div
        class="Card-Container-0"
      >
        <img
          alt="poster"
          class="Image-Container"
          src="abc.com"
        />
        <p
          class="Song-Name"
        >
          song_name
        </p>
        <span
          class="Artist-Name"
        >
          abc
        </span>
        <div
          class="Like-Button-Container"
        >
          <input
            alt="Like"
            class="Heart-Button"
            src="./assets/red-heart.svg"
            type="image"
          />
          <div
            class="centered"
          >
            1
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
